tests/test_autogen_diffs.py

```diff
@@ -693,6 +692,135 @@ def test_compare_type(
+        self.m1.create_all(self.bind)
+        m2 = MetaData()
+        Table("sometable", m2, Column("col", colb))
+
+        ctx_opts = {
+            "compare_type": True,
+            "compare_server_default": True,
+            "target_metadata": m2,
+            "upgrade_token": "upgrades",
+            "downgrade_token": "downgrades",
+            "alembic_module_prefix": "op.",
+            "sqlalchemy_module_prefix": "sa.",
+            "include_object": _default_object_filters,
+        }
+        if self.configure_opts:
+            ctx_opts.update(self.configure_opts)
+        with self.bind.connect() as conn:
+            context = MigrationContext.configure(
+                connection=conn, opts=ctx_opts
+            )
+            autogen_context = api.AutogenContext(context, m2)
+            uo = ops.UpgradeOps(ops=[])
+            autogenerate._produce_net_changes(autogen_context, uo)
+        return bool(uo.as_diffs())
+
+    @testing.combinations(
+        (INTEGER(),),
+        (CHAR(),),
+        (VARCHAR(32),),
+        (Text(),),
+        (FLOAT(),),
+        (Numeric(),),
+        (DECIMAL(),),
+        (TIMESTAMP(),),
+        (DateTime(),),
+        (Boolean(),),
+        (BigInteger(),),
+        (SmallInteger(),),
+        (DATE(),),
+        (String(32),),
+        (LargeBinary(),),
+        (Unicode(32),),
+        (Enum("one", "two", "three", name="the_enum"),),
+    )
+    def test_introspected_columns_match_metadata_columns(self, cola):
+        is_(self._compare_columns(cola, cola), False)
+
+    @testing.combinations(
+        (String(32), VARCHAR(32), False),
+        (VARCHAR(6), String(6), False),
+        (CHAR(), String(1), True),
+        (Text(), VARCHAR(255), True),
+        (Unicode(32), String(32), False, config.requirements.not_oracle),
+        (Unicode(32), VARCHAR(32), False, config.requirements.not_oracle),
+        (VARCHAR(6), VARCHAR(12), True),
+        (VARCHAR(6), String(12), True),
+    )
+    def test_string_comparisons(self, cola, colb, expect_changes):
+        is_(self._compare_columns(cola, colb), expect_changes)
+
+    @testing.combinations(
+        (
+            DateTime(),
+            DateTime(timezone=False),
+            False,
+            config.requirements.datetime_timezone,
+        ),
+        (
+            DateTime(),
+            DateTime(timezone=True),
+            True,
+            config.requirements.datetime_timezone,
+        ),
+        (
+            DateTime(timezone=True),
+            DateTime(timezone=False),
+            True,
+            config.requirements.datetime_timezone,
+        ),
+    )
+    def test_datetime_comparisons(self, cola, colb, expect_changes):
+        is_(self._compare_columns(cola, colb), expect_changes)
+
+    @testing.combinations(
+        (Integer(), Integer(), False),
+        (
+            Integer(),
+            Numeric(8, 0),
+            True,
+            config.requirements.integer_subtype_comparisons,
```

Comment:
**mike bayer** (zzzeek) wrote:

this requirement apparently refers to just "is oracle", where we couldn't compare BigInteger to Integer.  however, now that we compare the "scale" value, this should work now?

Reply:
Unfortunately not with the other rules. Integer and Numeric come through as synonyms on Oracle, and the "metadata" column has no arguments...so we ignore the arguments on the "inspect" column.